lib/Backend/arm64/LowerMD.cpp

```diff
@@ -3710,9 +3710,20 @@ LowererMD::GenerateFastShiftRight(IR::Instr * instrShift)
     else
     {
         // s2 = MOV src2
-        opndSrc2 = opndSrc2->UseWithNewType(TyInt32, this->m_func);
         opndReg = IR::RegOpnd::New(TyInt32, this->m_func);
-        instr = IR::Instr::New(Js::OpCode::MOV, opndReg, opndSrc2, this->m_func);
+
+        if (opndSrc2->IsImmediateOpnd())
+        {
+            Assert(opndSrc2->IsAddrOpnd() && opndSrc2->AsAddrOpnd()->IsVar());
+            opndSrc2 = IR::IntConstOpnd::New(Js::TaggedInt::ToInt32(opndSrc2->AsAddrOpnd()->m_address), TyInt32, this->m_func);
+            instr = IR::Instr::New(Js::OpCode::LDIMM, opndReg, opndSrc2, this->m_func);
+        }
+        else
+        {
+            opndSrc2 = opndSrc2->UseWithNewType(TyInt32, this->m_func);
+            instr = IR::Instr::New(Js::OpCode::MOV, opndReg, opndSrc2, this->m_func);
+        }
+
         instrShift->InsertBefore(instr);
```

Comment:
You should be able to use Lowerer::InsertMove here

Reply:
Yup and at other places too. Changed.

---
In reply to: [150096843](https://github.com/Microsoft/ChakraCore/pull/4189#discussion_r150096843) [](ancestors = 150096843)